odl/discr/lp_discr.py

```diff
@@ -1248,6 +1267,275 @@ def discr_sequence_space(shape, exponent=2.0, impl='numpy', **kwargs):
+
+    Examples
+    --------
+    >>> discr = uniform_discr([0, 0], [1, 2], (10, 5))
+    >>> discr.cell_sides
+    array([ 0.1,  0.4])
+
+    No additional arguments -> copy:
+
+    >>> uniform_discr_fromdiscr(discr) == discr
+    True
+
+    1 or 2 arguments -> pick missing information from discr.
+
+    Changing ``min_corner`` or ``max_corner`` translates the domain.
+    Changing ``cell_sides`` or ``nsamples`` keeps the domain and
+    adapts the cell sides:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1])
+    uniform_discr([1.0, 1.0], [2.0, 3.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, max_corner=[0, 0])
+    uniform_discr([-1.0, -2.0], [0.0, 0.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, cell_sides=[1, 1])
+    uniform_discr([0.0, 0.0], [1.0, 2.0], [1, 2])
+    >>> uniform_discr_fromdiscr(discr, nsamples=[5, 5])
+    uniform_discr([0.0, 0.0], [1.0, 2.0], [5, 5])
+    >>> uniform_discr_fromdiscr(discr, nsamples=[5, 5]).cell_sides
+    array([ 0.2,  0.4])
+
+    2 arguments -> pick missing information from discr.
+
+    Giving new ``min_corner`` and ``max_corner`` adapts the cell
+    sides, keeping the shape.
+    Combining one of ``min_corner`` and ``max_corner`` with ``nsamples``
+    computes the new domain by keeping ``cell_sides``.
+    Giving one of ``min_corner`` and ``max_corner`` together with
+    ``cell_sides`` keeps ``nsamples`` constant:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3])
+    uniform_discr([1.0, 1.0], [3.0, 3.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3]).cell_sides
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                nsamples=[5, 5])
+    uniform_discr([1.0, 1.0], [1.5, 3.0], [5, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                cell_sides=[1, 1])
+    uniform_discr([1.0, 1.0], [11.0, 6.0], [10, 5])
+
+    Specifying 3 or more new parameters will ignore ``discr``:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3],
+                                       cell_sides=[0.5, 0.25])
+    uniform_discr([1.0, 1.0], [3.0, 3.0], [4, 8])
+
+    Finally, everything can be done per axis. ``None`` entries are
+    interpreted as "not specified":
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[None, 1],
+    ...                                max_corner=[3, None],
+                                       cell_sides=[None, 0.25])
+    uniform_discr([2.0, 1.0], [3.0, 2.25], [10, 5])
+    """
+    if not isinstance(discr, DiscreteLp):
+        raise TypeError('`discr` {!r} is not a DiscreteLp instance'
+                        ''.format(discr))
+    if not discr.partition.is_regular:
+        raise ValueError('`discr` {} is not uniformly discretized'
```

Comment:
Well then we honestly may want to do a rename of the `is_regular` thing to `is_uniform`? For clarity.